libs/langchain-community/src/vectorstores/cassandra.ts

```diff
@@ -649,4 +730,113 @@ export class CassandraStore extends VectorStore {
     // Wait for all insert operations to complete.
     await Promise.all(insertPromises);
   }
+
+  /**
+   * Method to get the CassandraClient
+   * @param argsWithDefaults connection arguments
+   * @returns CassandraClient object
+   */
+  static async getClient(args: ClientArgs): Promise<CassandraClient> {
+    if (!args.serviceProviderArgs) {
+      return new CassandraClient(args);
+    }
+
+    if (args.serviceProviderArgs && args.serviceProviderArgs.astra) {
+      return CassandraStore.getAstraClient(args);
+    }
+
+    throw new Error("Unsupported configuration for Cassandra client.");
+  }
+
+  static async getAstraClient(args: ClientArgs): Promise<CassandraClient> {
+    const astraConfig = args.serviceProviderArgs?.astra;
+    if (!astraConfig) {
+      throw new Error(
+        "Astra configuration is not provided in serviceProviderArgs."
+      );
+    }
+
+    // Create a copy of args to avoid direct modification
+    const modifiedArgs = { ...args };
+
+    // Initialize cloud if it's not already defined
+    modifiedArgs.cloud = modifiedArgs.cloud || { secureConnectBundle: "" };
+
+    if (!modifiedArgs.cloud.secureConnectBundle) {
+      // Store the secureConnectBundle in a temporary file
+      const dir = path.join(os.tmpdir(), "cassandra-astra");
+      await fs.mkdir(dir, { recursive: true });
+
+      let scbFileName = `astra-secure-connect-${astraConfig.datacenterID}`;
+      if (astraConfig.regionName) {
+        scbFileName += `-${astraConfig.regionName}`;
+      }
+      scbFileName += ".zip";
+      const scbPath = path.join(dir, scbFileName);
+
+      // Try accessing the file, if it doesn't exist, download it
+      try {
+        await fs.access(scbPath);
+      } catch (error: unknown) {
+        // Handle file not found error (ENOENT)
+        if (typeof error === "object" && error !== null && "code" in error) {
+          // Download secure connect bundle
+          const bundleURLTemplate =
+            "https://api.astra.datastax.com/v2/databases/{database_id}/secureBundleURL?all=true";
+          const url = bundleURLTemplate.replace(
+            "{database_id}",
+            astraConfig.datacenterID
+          );
+          const postResponse = await fetch(url, {
+            method: "POST",
+            headers: {
+              Authorization: `Bearer ${astraConfig.token}`,
```

Comment:
Meta question: if you can fetch the bundle using an auth token and the bundle doesn't expire, what is the purpose of authenticating with the bundle instead of just skipping a step and using a token directly?

Reply:
The AstraDB mode works in that way - it's a RESTful service. In the Cassandra side (when we connect via the driver), the bundle includes a .json file with some metadata, and also artifacts to support the mTLS connection requirements.

Now, these are signed with a 10 year key...so they do expire. I'll add some logic here to keep the file current.